src/hera/workflows/parameter.py

```diff
@@ -20,6 +20,13 @@ class Parameter(_ModelParameter):
     for Steps and Tasks to assign values.
     """
 
+    name: Optional[str] = None  # type: ignore
+
+    def check_name(self):
```

Comment:
Should be a private function
```suggestion
    def _check_name(self):
```